ably/realtime_conn_spec_test.go

```diff
@@ -2358,3 +2358,97 @@ func TestRealtimeConn_RTN19a(t *testing.T) {
 		t.Fatalf("expected %v got %v", expect, got)
 	}
 }
+
+func TestRealtimeConn_RTN24_RTN21_RTC8a_RTN4h_Override_ConnectionDetails_On_Connected(t *testing.T) {
+	t.Parallel()
+
+	in := make(chan *proto.ProtocolMessage, 1)
+	out := make(chan *proto.ProtocolMessage, 16)
+
+	c, _ := ably.NewRealtime(
+		ably.WithAutoConnect(false),
+		ably.WithToken("fake:token"),
+		ably.WithDial(ablytest.MessagePipe(in, out)),
+	)
+
+	connDetails := proto.ConnectionDetails{
+		ClientID:           "id1",
+		ConnectionKey:      "foo",
+		MaxFrameSize:       12,
+		MaxInboundRate:     14,
+		MaxMessageSize:     67,
+		ConnectionStateTTL: proto.DurationFromMsecs(time.Minute * 2),
+		MaxIdleInterval:    proto.DurationFromMsecs(time.Second),
+	}
+
+	in <- &proto.ProtocolMessage{
+		Action:            proto.ActionConnected,
+		ConnectionID:      "connection-id-1",
+		ConnectionDetails: &connDetails,
+	}
+
+	err := ablytest.Wait(ablytest.ConnWaiter(c, c.Connect, ably.ConnectionEventConnected), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	newConnDetails := proto.ConnectionDetails{
+		ClientID:           "id2",
+		ConnectionKey:      "bar",
+		MaxFrameSize:       13,
+		MaxInboundRate:     15,
+		MaxMessageSize:     70,
+		ConnectionStateTTL: proto.DurationFromMsecs(time.Minute * 2),
+		MaxIdleInterval:    proto.DurationFromMsecs(time.Second),
+	}
+
+	// Re-attach the channel
```

Comment:
```suggestion
	// Send new connection details
```

But I would move this comment to line 2416 or 2395, or just remove it.